service-api/module/Application/src/Yoti/SessionStatusService.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Yoti;
+
+use Application\Fixtures\DataImportHandler;
+use Application\Model\Entity\CaseData;
+use Application\Model\Entity\CounterService;
+use Application\Yoti\Http\Exception\YotiException;
+use DateTime;
+use InvalidArgumentException;
+use Ramsey\Uuid\Uuid;
+
+class SessionStatusService
+{
+    public function __construct(
+        public readonly YotiService $yotiService,
+        public readonly DataImportHandler $dataImportHandler
+    ) {
+    }
+
+    public function getSessionStatus(CaseData $caseData): string|array|CounterService
```

Comment:
I thought the whole point of this class would be to prevent unnecessary calls to Yoti if we can derive current status by checking first with notification updates from db. So it would need to return something if the controller was to subsequently use that to fetch the actual results?

Reply:
If you used `PutItem` for the updates, the controller could just read from `CaseData` since it will have the latest information on it. Otherwise I think it should probably return `CounterService` since that contains the data points the controller would need.

Reply:
Ah ok that makes more sense. Will just add logs for any errors in this case :)